serverlessworkflow/modules/ROOT/pages/persistence/postgresql-flyway-migration.adoc

```diff
@@ -0,0 +1,101 @@
+
+=== Migrate using Flyway CLI
+* If you want to migrate manually you can use the Flyway migration CLI tool.
+
+** Download Flyway migration command-line tool from the link:{flyway_install_url}[Download Flyway Community Edition] link.
+** Add the following properties in the `/conf/flyway.conf` file:
++
+--
+[source,console]
+----
+flyway.url=jdbc:postgresql://localhost:5432/foobardb
+flyway.user=foo
+flyway.password=bar
+----
+--
+** You can specify these options with commands as well. The Flyway CLI will prompt for the username and password if they are missing in the configuration.
++
+--
+[source,console]
+----
+$ flyway migrate -url=jdbc:postgresql://localhost:5432/foobardb -user=foo -password=bar
+----
+--
+** Download the Kogito DDL zip from link:{kogito_ddl_script_url}[this link], choose the correct version,  extract the files and use the scripts from the `postgresql` folder.
+** You can specify the location of the SQL files that need to be migrated using the `flyway.locations` option. For example,
++
+--
+[source,console]
+----
+flyway.locations=classpath:com.mycomp.migration,database/migrations,filesystem:/sql-migrations,s3:migrationsBucket,gcs:migrationsBucket
+----
+--
+
+=== Manually executing scripts
+You can use the provided SQL scripts in the zip file to migrate the database by executing them one by one.
+
+* You can find the Kogito DDL scripts by visiting link:{kogito_ddl_script_url}[this link].
+* Download the zip file of the scripts from the needed version for the Kogito upgrade.
+* Extract the files.
+* Execute the `sql` files from `postgresql` folder.
++
+--
+[source,console]
+----
+> psql -H host -U username -d database_name -a -f create_table.sql
+----
+--
+
+== Baseline migration
+In order to apply migrations to an existing schema. It is necessary to establish the baseline from where Flyway should start executing new migrations. That means, in case your schema already contains a structure and from now on, new changes provided by Kogito should be applied via Flyway integration. In Flyway terms, only migrations above `baselineVersion` will then be applied.
+Using below properties you can initialize flyway schema table.
+
+=== Baseline using Quarkus Configuration:
+--
+[source,console]
+----
+quarkus.flyway.baseline-on-migrate=true
+quarkus.flyway.baseline-version=1.32.0
+----
+--
+=== Baseline using Flyway CLI:
+--
+[source,console]
+----
+$ flyway -baselineOnMigrate="true" -baselineVersion="1.32.0" migrate
+----
+--
+
+For more details regarding Baseline migration, please visit the link:{flyway_baseline_migration_url}[Flyway Documentation].
+
+NOTE: Be careful when enabling this feature as it removes the safety net which ensures that Flyway does not migrate the wrong database in case of a configuration mistake.
+
+For more details regarding existing the Database setup, visit the link:{flyway_migrate_existing_url}[Flyway Documentation].
```

Comment:
```suggestion
== Baseline migration

In order to apply migrations to an existing schema. It is necessary to establish the baseline from where Flyway should start executing new migrations. That means, in case your schema already contains a structure and from now on, new changes provided by Kogito should be applied via Flyway integration. In Flyway terms, only migrations above `baselineVersion` will then be applied.
Using below properties you can initialize flyway schema table.

WARNING: Be careful when enabling this feature as it removes the safety net which ensures that Flyway does not migrate the wrong database in case of a configuration mistake.

=== Baseline using Quarkus Configuration
--
[source,console]
----
quarkus.flyway.baseline-on-migrate=true
quarkus.flyway.baseline-version=1.32.0
----
--
=== Baseline using Flyway CLI
--
[source,console]
----
$ flyway -baselineOnMigrate="true" -baselineVersion="1.32.0" migrate
----
--

=== More details on Flyway migrations

* link:{flyway_baseline_migration_url}[Baseline Migrations Documentation].
* link:{flyway_migrate_existing_url}[Existing Database Setup Documentation].
```